jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -268,7 +268,7 @@ private SpanContext createNewContext(String debugId) {
 
       byte flags = 0;
       if (debugId != null) {
-        flags |= SpanContext.flagSampled | SpanContext.flagDebug;
+        flags = (byte) (flags | SpanContext.flagSampled | SpanContext.flagDebug);
```

Comment:
all operands are already of type `byte`, why do we need to re-cast?

Reply:
Because of assignment conversion. See http://docs.oracle.com/javase/specs/jls/se7/html/jls-5.html#jls-5.2 and http://errorprone.info/bugpattern/NarrowingCompoundAssignment

The result of the bitwise OR is being promoted to a `int` which was previously implicitly cast to a `byte`.